CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportTopComponent.java

```diff
@@ -0,0 +1,894 @@
+        
+        updateFilterData();
+    }
+    
+    public void updateFilterData(){
+        popupFilters.clear();
+        if (severePopCheckBox.isSelected()) {
+            popupFilters.add("SEVERE");
+        }
+        if (warningPopCheckBox.isSelected()) {
+            popupFilters.add("WARNING");
+        }
+        if (infoPopCheckBox.isSelected()) {
+            popupFilters.add("INFO");
+        }
+        if (finePopCheckBox.isSelected()) {
+            popupFilters.add("FINE");
+        }
+    }
+    
+    private void updateSettingsIcon(final FlowPane settingsPane, final String innerShade, final String borderShade){
+        settingsPane.setStyle("-fx-background-color: " + innerShade + "; -fx-border-color: " + borderShade + ";");
+    }
+    
+    public void setReportsExpanded(final boolean expandedMode) {
+        final int errCount = sessionErrors.size();
+        if (sessionErrorsBox.getChildren().size() > 0) {
+            for (int i = 0; i < errCount; i++) {
+                ((TitledPane) sessionErrorsBox.getChildren().get(i)).setExpanded(expandedMode);
+            }
+        }
+    }
+
+    public ErrorReportEntry findActiveEntryWithId(final double id) {
+        for (final ErrorReportEntry activeEntry : sessionErrors) {
+            if (activeEntry.getEntryId() == id) {
+                return activeEntry;
+            }
+        }
+        return null;
+    }
+
+    public void refreshSessionErrors() {
+        // save expanded states
+        final int errCount = sessionErrors.size();
+        if (sessionErrorsBox.getChildren().size() > 0) {
+            for (int i = 0; i < errCount; i++) {
+                ErrorReportSessionData.getInstance().updateDisplayedEntryScreenSettings(
+                        sessionErrors.get(i).getEntryId(), null, null, ((TitledPane) sessionErrorsBox.getChildren().get(i)).isExpanded());
+            }
+        }
+        final ArrayList<String> activeFilters = new ArrayList<>();
+        if (severeRepCheckBox.isSelected() || severePopCheckBox.isSelected()) {
+            activeFilters.add("SEVERE");
+        }
+        if (warningRepCheckBox.isSelected() || warningPopCheckBox.isSelected()) {
+            activeFilters.add("WARNING");
+        }
+        if (infoRepCheckBox.isSelected() || infoPopCheckBox.isSelected()) {
+            activeFilters.add("INFO");
+        }
+        if (fineRepCheckBox.isSelected() || finePopCheckBox.isSelected()) {
+            activeFilters.add("FINE");
+        }
+        final ArrayList<ErrorReportEntry> combinedErrors = ErrorReportSessionData.getInstance().refreshDisplayedErrors(activeFilters);
+        sessionErrors.clear();
+        hiddenErrors.clear();
+        for (ErrorReportEntry entry : combinedErrors) {
+            switch (entry.getErrorLevel().getName()) {
+                case "SEVERE":  if (severeRepCheckBox.isSelected()) {
```

Comment:
given their use across the PR, maybe some constants for the logging levels would make sense?